### ISSUE TITLE
Multiple Max deployments with single entry point

### ISSUE THREAD

**Issue by joseluisrc** [Azure/Moodle#199]
Hello all.

The Max deployment script handles about 5000+ concurrent users. How would it be possible to increase the number of concurrent users to 30K+?

Does increasing the minimum of instances in the Virtual machine scale set allow for more concurrent users? Or would a different approach be needed, i.e. deploy multiple Max installs and somehow use a load balancer as a single entry point? In that case, can the multiple Max deployments use the same DB backend?

Any help is appreciated.

**Comment by UmakanthOS** [comment]
Hello, that's something we're looking at. Depending on the nature of your deployment, we'll likely run into I/O or DB bottlenecks. As for CPU, resizing the VMSS instances from DS2v2 to higher sized instances and/or setting scaling rules should address requirements. Would you be able to share more about your requirements? (in addition to 30K concurrent users?)

**Comment by joseluisrc** [comment]
Thanks, UmakanthOS.

The problem is that I deployed the Large Moodle installation and it failed with just 2000 students. I tried to manually scale the vmss but that didn't correct the situation. Nobody could get into the quiz. The most common errors where: 500 and 503 (varnish caché). Since it was a "live" quiz, there wasn't much I could do.

I can't discard a DDOS attack, but I'm also considering that the load balancer failed, as a single entry point that it is.

Since the Large installation failed with only 2000 "real" students, that's why I was thinking that multiple Max installations would help to prevent this from happening again. I'm also researching if multiple load balancers are possible in Azure.

Thanks!
